eng/BuildTask.targets

```diff
@@ -0,0 +1,118 @@
+    <PackageReference Update="NuGet.Packaging" Publish="false" />
+    <PackageReference Update="NuGet.ProjectModel" Publish="false" />
+    <PackageReference Update="NuGet.Versioning" Publish="false" />
+  </ItemGroup>
+
+  <!-- Don't include assemblies that are inbox in Desktop MSBuild -->
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <PackageReference Update="System.Buffers" Publish="false" />
+    <PackageReference Update="System.Memory" Publish="false" />
+    <PackageReference Update="System.Numerics.Vectors" Publish="false" />
+    <PackageReference Update="System.Reflection.Metadata" Publish="false" />
+    <PackageReference Update="System.Reflection.MetadataLoadContext" Publish="false" />
+    <PackageReference Update="System.Runtime.CompilerServices.Unsafe" Publish="false" />
+    <PackageReference Update="System.Security.Cryptography.Xml" Publish="false" />
+    <PackageReference Update="System.Text.Encodings.Web" Publish="false" />
+    <PackageReference Update="System.Text.Json" Publish="false" />
+    <PackageReference Update="System.Threading.Tasks.Dataflow" Publish="false" />
+    <PackageReference Update="System.Threading.Tasks.Extensions" Publish="false" />
+    <PackageReference Update="System.ValueTuple" Publish="false" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <!--
+      Update all PackageReference items to default Publish to true.
+      This forces the publish output to contain the dlls.
+     -->
+    <PackageReference Update="@(PackageReference)">
+      <Publish Condition="'%(PackageReference.Publish)' == ''">true</Publish>
+      <PrivateAssets Condition="'%(PackageReference.Publish)' == 'false'">all</PrivateAssets>
+    </PackageReference>
+
+    <!--
+      Update all Reference items to have Pack="false"
+      This removes the frameworkDependency nodes from the generated nuspec
+    -->
+    <Reference Update="@(Reference)"
+               Pack="false" />
+  </ItemGroup>
+
+  <!-- Desktop MSBuild compatibilty -->
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <PackageReference Update="System.Text.Json" Version="7.0.1" />
+    <PackageReference Update="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
+    <PackageReference Update="Microsoft.Extensions.DependencyModel" Version="7.0.0" />
+    <PackageReference Update="System.Collections.Immutable" Version="7.0.0" />
+    <PackageReference Update="System.Reflection.Metadata" Version="7.0.0" />
+  </ItemGroup>
+
+  <!-- Publish .NET Core assets and include them in the package under $(BuildTaskTargetFolder) directory. -->
+  <Target Name="_AddBuildOutputToPackageCore" DependsOnTargets="Publish" Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+    <PropertyGroup>
+      <BuildTaskTargetTfmSpecificFolder Condition="'$(BuildTaskTargetTfmSpecificFolder)' == ''">net</BuildTaskTargetTfmSpecificFolder>
+    </PropertyGroup>
```

Comment:
We shouldn't need this here, let's hard-code "net" and not make it customizable.

Reply:
Fixed